code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,309 @@
+
+	user.unset_interaction()
+	gun.deployed = FALSE
+	gun.deploy_integrity = obj_integrity
+
+	user.put_in_active_hand(gun)
+	gun = null
+
+	qdel(src)
+
+///Makes sure we clean our trash
+/obj/machinery/mounted/Destroy()
+	if(gun)
+		qdel(gun)
+	if(operator)
+		operator.unset_interaction()
+	. = ..()
+
+///Left in from old TL-102 code
+/obj/machinery/mounted/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE) // Those Ayy lmaos.
+	SEND_SIGNAL(X, COMSIG_XENOMORPH_ATTACK_M56)
+	return ..()
+
+///Update health hud when damage is taken
+/obj/machinery/mounted/take_damage(damage_amount, damage_type, damage_flag, effects, attack_dir, armour_penetration)
+	. = ..()
+	hud_set_machine_health()
+
+///This is called when a user tries to operate the gun
+/obj/machinery/mounted/interact(mob/user)
+	if(!ishuman(user))
+		return TRUE
+	var/mob/living/carbon/human/human_user = user
+	if(human_user.interactee == src) //If your already using the gun, and click on it again to exit
+		human_user.unset_interaction()
+		visible_message("[icon2html(src, viewers(src))] <span class='notice'>[human_user] decided to let someone else have a go. </span>",
+			"<span class='notice'>You decided to let someone else have a go on the [src] </span>")
+		return TRUE
+	if(get_step(src, REVERSE_DIR(dir)) != human_user.loc) //cant man the gun from the barrels side
+		to_chat(human_user, "<span class='warning'>You should be behind [src] to man it!</span>")
+		return TRUE
+	if(operator) //If there is already a operator then they're manning it.
+		if(!operator.interactee)
+			stack_trace("/obj/machinery/mounted/interact called by user [human_user] with an operator with a null interactee: [operator].")
+			operator = null //this shouldn't happen, but just in case
+		else
+			to_chat(human_user, "<span class='warning'>Someone's already controlling it.</span>")
+			return TRUE
+	if(human_user.interactee) //Make sure we're not manning two guns at once, tentacle arms.
+		to_chat(human_user, "<span class='warning'>You're already busy!</span>")
+		return TRUE
+	if(issynth(human_user) && !CONFIG_GET(flag/allow_synthetic_gun_use))
+		to_chat(human_user, "<span class='warning'>Your programming restricts operating heavy weaponry.</span>")
+		return TRUE
+
+	visible_message("[icon2html(src, viewers(src))] <span class='notice'>[human_user] mans the [src]!</span>",
+		"<span class='notice'>You man the gun!</span>")
+
+	operator = human_user
+
+	return ..()
+
+/obj/machinery/mounted/examine(mob/user) //Let us see how much ammo we got in this thing.
+	. = ..()
+	gun.examine(user)
+
+/obj/machinery/mounted/attackby(obj/item/I, mob/user, params) //This will be how we take it apart.
+	. = ..()
+
+	if(!ishuman(user))
+		return
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return
```

Comment:
Seems terrible code, you took it from somewhere?

Reply:
Yes, its from old TL-102 code